### ISSUE TITLE
[C++] a < comparison followed by a >= comparison is tokenized wrongly

### ISSUE THREAD

**Issue by Tralomine** [atom/language-c#326]
### Description

a < comparison followed by a >= comparison is tokenized wrongly

### Steps to Reproduce

1. try typing
```C
(x < x || x >= x)
```
2. ???
3. profit

![image](https://user-images.githubusercontent.com/11725390/57316846-08e57800-70f7-11e9-91c3-71dc1a468ad7.png)

**Expected behavior:** the comparisons operators should be all the same color and in only one token

**Actual behavior:** they are split and considered as 
```html
<span class="syntax--punctuation syntax--definition syntax--template syntax--bracket syntax--angle">&gt;</span>
<span class="syntax--keyword syntax--operator">=</span>
```

note: trying with multiples comparisons in this style always end with only the last one experiencing this behaviour

![image](https://user-images.githubusercontent.com/11725390/57317416-557d8300-70f8-11e9-8ae4-24fd43e94215.png)

note: the first variable in the comparison is considered as
```html
<span class="syntax--entity syntax--name syntax--function">x</span>
```
(and therefore is blue)

**Comment by 50Wliu** [comment]
Reproduced with the C++ tree-sitter grammar but not the C one.

**Comment by Ben3eeE** [comment]
Tree sitter parses this as:

template_function
template_argument_list

Because there is a starting and ending angle bracket in the statement.

Which is scoped as `punctuation.definition.template.bracket.angle` because it isn't an operator when it's a template definition here:
https://github.com/atom/language-c/blob/a355a4d4a019d5684c40072b81f24c89e58dc56f/grammars/tree-sitter-cpp.cson#L198-L201


The rule for `>` and `<` is only added to `relational_expression` to not conflict with templates:
https://github.com/atom/language-c/blob/a355a4d4a019d5684c40072b81f24c89e58dc56f/grammars/tree-sitter-cpp.cson#L224-L225

**Comment by maxbrunsfeld** [comment]
Thanks for the report! The fix will ship in Atom 1.38 (and 1.38-beta0).
